### PR TITLE
chore(table): refactor components

### PR DESCRIPTION
**Pull request by joseazevedo88**
- Remove react-aria from table components
- Improve keyboard navigation, both checkboxes and action buttons can now be navigated to and interacted with using tabs and space(validated with Jamie)
- Improve code by removing the child components dependency on the huge state object created in the <Table> component.

### CODE REVIEW

File: src/components/Table/Table.tsx [hunk 1]
```diff
@@ -44,17 +43,19 @@ export type Row = {
...
     selectedRowIds?: (string | number)[];
     ariaLabel?: string;
 }>;
 
 const DEFAULT_SORT_ORDER = 'descending';
 
+export type SortDirection = 'ascending' | 'descending';
+
 type SortType = {
     sortedColumnKey?: string | number;
```
Comment (imoutaharik) [inline src/components/Table/Table.tsx:57]: It would make sense to use the `Key` type here as well. Also on type `Row` `key: Key` and other places. don't you think?

File: src/components/Table/Table.tsx [hunk 2]
```diff
@@ -118,52 +119,57 @@ export const Table = ({
...
+                                        handleSortChange={onSortChange}
+                                        setSelectedRows={onSelectionChange}
                                     />
-                                ) : (
-                                    <TableColumnHeader key={column.key} column={column} state={state} />
-                                ),
-                            )}
+                                );
+                            })}
+                            <td />
```
Comment (imoutaharik) [inline src/components/Table/Table.tsx:172]: I think that closing `td` can be removed.

File: src/components/Table/TableCell.tsx [hunk 3]
```diff
@@ -17,50 +11,69 @@ export enum TableCellType {
...
-    state: TableState<any>;
+    selectionMode: string;
     type?: TableCellType;
+    isChecked?: boolean;
+    selectedRows: Key[];
+    setSelectedRows?: (ids?: Key[]) => void;
 };
 
-export const TableCell: FC<TableCellProps> = ({ cell, state, type = TableCellType.Default }) => {
+export const TableCell: FC<TableCellProps> = ({
```
Comment (imoutaharik) [inline src/components/Table/TableCell.tsx:21]: `FC` can be removed here as well.

File: src/components/Table/TableCell.tsx [hunk 4]
```diff
@@ -17,50 +11,69 @@ export enum TableCellType {
...
+            if (!setSelectedRows) {
+                return;
+            }
+
+            if (isChecked) {
+                const filteredRows = selectedRows.filter((row) => row !== cell.parentKey);
+                setSelectedRows(filteredRows);
+                return;
+            }
+            if (selectionMode === SelectionMode.SingleSelect) {
```
Comment (imoutaharik) [inline src/components/Table/TableCell.tsx:43]: That last condition could be simplified, WDYT?
```
const selectRows = selectionMode === SelectionMode.SingleSelect ? [cell.parentKey] : [...selectedRows, cell.parentKey];
setSelectedRows(selectRows);
```
giving the variable a better name of course. 😄

File: src/components/Table/TableCell.tsx [hunk 5]
```diff
@@ -17,50 +11,69 @@ export enum TableCellType {
...
                 ref={ref}
                 className={merge([
                     'tw-pl-8 tw-py-4 tw-pr-4 tw-border-l-4',
-                    checked ? 'tw-border-violet-60' : 'tw-border-transparent',
+                    isChecked ? 'tw-border-violet-60' : 'tw-border-transparent',
                 ])}
                 data-test-id="table-select-cell"
             >
-                <Checkbox
+                <CheckboxCmp
```
Comment (imoutaharik) [inline src/components/Table/TableCell.tsx:60]: I'm curious, why was this renamed?

File: src/components/Table/TableCell.tsx [hunk 6]
```diff
@@ -17,50 +11,69 @@ export enum TableCellType {
...
     return (
         <td
-            {...mergeProps(gridCellProps, focusProps)}
+            role="cell"
             ref={ref}
             className={merge([
                 'tw-p-4 tw-font-normal tw-text-xs focus:tw-outline-none',
-                checked ? 'tw-text-black-100 dark:tw-text-white' : 'tw-text-black-80 dark:tw-text-black-20',
-                isFocusVisible && FOCUS_STYLE_INSET,
+                isChecked ? 'tw-text-black-100 dark:tw-text-white' : 'tw-text-black-80 dark:tw-text-black-20',
```
Comment (imoutaharik) [inline src/components/Table/TableCell.tsx:76]: the classes here should be changed to use `Alias tokens` instead and avoid to use `dark` prefix.
We shouldn't have things like `tw-text-black-100` for example or violet-...

File: src/components/Table/TableColumnHeader.tsx [hunk 7]
```diff
@@ -20,100 +14,105 @@ export enum TableColumnHeaderType {
...
     type?: TableColumnHeaderType;
+    rowIds: Key[];
+    sortDirection?: SortDirection;
+    selectionMode: string;
+    isColumnSorted?: boolean;
+    handleSortChange: (column: Key, direction: SortDirection) => void;
+    setSelectedRows?: (ids?: Key[]) => void;
 };
 
 export const TableColumnHeader: FC<TableColumnHeaderProps> = ({
```
Comment (imoutaharik) [inline src/components/Table/TableColumnHeader.tsx:26]: Same thing here `FC` can be removed.

File: src/components/Table/TableHeaderRow.tsx [hunk 8]
```diff
@@ -1,19 +1,10 @@
...
-import { TableState } from '@react-stately/table';
 import React, { FC, useRef } from 'react';
 
-export type TableHeaderRowProps = {
-    item: any;
-    state: TableState<any>;
-};
-
-export const TableHeaderRow: FC<TableHeaderRowProps> = ({ item, state, children }) => {
+export const TableHeaderRow: FC = ({ children }) => {
```
Comment (imoutaharik) [inline src/components/Table/TableHeaderRow.tsx:3]: Another `FC`

File: src/components/Table/TableRow.tsx [hunk 9]
```diff
@@ -1,30 +1,22 @@
...
 import React, { FC, useRef } from 'react';
 
 export type TableRowProps = {
-    item: any;
-    state: TableState<any>;
+    isSelected?: boolean;
 };
 
-export const TableRow: FC<TableRowProps> = ({ item, state, children }) => {
+export const TableRow: FC<TableRowProps> = ({ isSelected = false, children }) => {
```
Comment (imoutaharik) [inline src/components/Table/TableRow.tsx:7]: Same here with the `FC`

**Review by imoutaharik — CHANGES_REQUESTED** [review]

**Comment by imoutaharik** [comment]
Nice!! 🎉  I'm happy to see that component free of React-aria. I left a few comments here and there but it looks good overall.
I'm just not sure about the keyboard navigation, you can only select the checkbox by using `space`, I was trying to do it with `Enter`, that made more sense to me. 🤔

File: src/components/Table/Table.tsx [hunk 10]
```diff
@@ -44,17 +43,19 @@ export type Row = {
...
-    onSelectionChange?: (ids?: (string | number)[]) => void;
+    onSelectionChange?: (ids?: Key[]) => void;
     selectionMode?: SelectionMode;
     selectedRowIds?: (string | number)[];
     ariaLabel?: string;
 }>;
 
 const DEFAULT_SORT_ORDER = 'descending';
 
+export type SortDirection = 'ascending' | 'descending';
```
[the remark this answers is not in the document]
Comment (joseazevedo88) [reply src/components/Table/Table.tsx:54]: Sounds good 👍

File: src/components/Table/Table.tsx (hunk 1 above)
Comment (joseazevedo88) [reply src/components/Table/Table.tsx:57]: Total sense 👍

File: src/components/Table/TableCell.tsx (hunk 5 above)
Comment (joseazevedo88) [reply src/components/Table/TableCell.tsx:60]: I also don't like it :( But it's a condition in the quality checks to not have sonarLint warnings and there was a warning in the `Checkbox` enum value because something called `Checkbox` already was being imported.

File: src/components/Table/TableCell.tsx (hunk 4 above)
Comment (joseazevedo88) [reply src/components/Table/TableCell.tsx:43]: Sure 👍

File: src/components/Table/TableCell.tsx (hunk 5 above)
Comment (joseazevedo88) [reply src/components/Table/TableCell.tsx]: Renamed to `CheckboxComponent` 👍

File: src/components/Table/TableCell.tsx (hunk 6 above)
Comment (joseazevedo88) [reply src/components/Table/TableCell.tsx:75]: Will be resolved in another PR as per discussion in slack
